### ISSUE TITLE
`DW_EH_PE_omit` not handled in `.eh_frame_hdr`, leading to `Unknown Encoding` error

### ISSUE THREAD

**Issue by Zhiwuyazhe1** [revng/revng#408]
## background

In [Ground Truth for Binary Disassembly is Not Easy](https://www.usenix.org/conference/usenixsecurity22/presentation/pang-chengbin), they perform a study on the approaches to building the ground truth for binary disassembly.We want to use revng to test on the ground truth they performed, but we ran into some problems.

[OracleGT github](https://github.com/junxzm1990/x86-sok) is their open source tool. They modified gcc and llvm to extract ground truth information of basic blocks, functions, and jump tables during the compilation process.

The following is the source code I compiled and the compilation process (all based on arm32):

arm32 OracleGT docker : `docker pull z472421519/arm32_gt`

hello_world source code:
```c
#include <stdio.h>

int main() {
  printf("hello world\n");
}
```
environment variable set(in docker): `root@32380fe55c2a:/gt_arm32# source gcc32_arm.rc`

dynamic compile: `$CC $CFLAGS $LDFLAGS -o dynamic_hello hello_world.c`

static compile: `$CC $CFLAGS $LDFLAGS -static -o static_hello hello_world.c`

We will provide the binary as an attachment.

We would like to know how to solve the following problem.

## issue

`revng model import binary dynamic_hello`

When I tried the above command, revng produced the following error report:

```
Abort at /builds/gitlab/revng/orchestra/orchestra/sources/revng/lib/Model/Importer/Binary/DwarfReader.h:246:

Unknown Encoding
PLEASE submit a bug report to https://github.com/revng/revng and include the crash backtrace
Stack dump:
0.      Program arguments: /home/bruce/orchestra/root/libexec/revng/revng-model-import-binary dynamic_hello
 #0 0x00007fcea2e816ce llvm::sys::PrintStackTrace(llvm::raw_ostream&, int) /builds/gitlab/revng/orchestra/orchestra/sources/llvm-project/llvm/lib/Support/Unix/Signals.inc:570:13
 #1 0x00007fcea2e7fa34 llvm::sys::RunSignalHandlers() /builds/gitlab/revng/orchestra/orchestra/sources/llvm-project/llvm/lib/Support/Signals.cpp:105:18
 #2 0x00007fcea2e81c3d SignalHandler(int) /builds/gitlab/revng/orchestra/orchestra/sources/llvm-project/llvm/lib/Support/Unix/Signals.inc:415:1
 #3 0x00007fcea67c5420 __restore_rt (/lib/x86_64-linux-gnu/libpthread.so.0+0x14420)
 #4 0x00007fcea297500b raise (/lib/x86_64-linux-gnu/libc.so.6+0x4300b)
 #5 0x00007fcea2954859 abort (/lib/x86_64-linux-gnu/libc.so.6+0x22859)
 #6 0x00007fcea7346059 /builds/gitlab/revng/orchestra/orchestra/sources/revng/lib/Support/Assert.cpp:18:3
 #7 0x00007fcea73460b8 (/home/bruce/orchestra/root/libexec/revng/../../////////////////////////////////////////////////////////////////////////////////lib/librevngSupport.so+0x240b8)
 #8 0x00007fcea7913d46 (/home/bruce/orchestra/root/libexec/revng/../../////////////////////////////////////////////////////////////////////////////////lib/revng/analyses/librevngModelImporterBinary.so+0x90d46)
 #9 0x00007fcea78d0582 DwarfReader<llvm::object::ELFType<(llvm::support::endianness)1, false>>::readUnsignedValue(unsigned int) /builds/gitlab/revng/orchestra/orchestra/sources/revng/lib/Model/Importer/Binary/DwarfReader.h:146:34
#10 0x00007fcea78d0582 (anonymous namespace)::ELFImporter<llvm::object::ELFType<(llvm::support::endianness)1, false>, false>::ehFrameFromEhFrameHdr() /builds/gitlab/revng/orchestra/orchestra/sources/revng/lib/Model/Importer/Binary/ELFImporter.cpp:860:41
#11 0x00007fcea78d0582 (anonymous namespace)::ELFImporter<llvm::object::ELFType<(llvm::support::endianness)1, false>, false>::import(ImporterOptions const&) /builds/gitlab/revng/orchestra/orchestra/sources/revng/lib/Model/Importer/Binary/ELFImporter.cpp:272:36
#12 0x00007fcea78c36ee std::__1::default_delete<(anonymous namespace)::ELFImporterBase>::operator()[abi:v160001]((anonymous namespace)::ELFImporterBase*) const /builds/gitlab/revng/orchestra/orchestra/root/lib64/llvm/clang-release/bin/../include/c++/v1/__memory/unique_ptr.h:65:5
#13 0x00007fcea78c36ee std::__1::unique_ptr<(anonymous namespace)::ELFImporterBase, std::__1::default_delete<(anonymous namespace)::ELFImporterBase>>::reset[abi:v160001]((anonymous namespace)::ELFImporterBase*) /builds/gitlab/revng/orchestra/orchestra/root/lib64/llvm/clang-release/bin/../include/c++/v1/__memory/unique_ptr.h:297:7
#14 0x00007fcea78c36ee std::__1::unique_ptr<(anonymous namespace)::ELFImporterBase, std::__1::default_delete<(anonymous namespace)::ELFImporterBase>>::~unique_ptr[abi:v160001]() /builds/gitlab/revng/orchestra/orchestra/root/lib64/llvm/clang-release/bin/../include/c++/v1/__memory/unique_ptr.h:263:75
#15 0x00007fcea78c36ee importELF(TupleTree<model::Binary>&, llvm::object::ELFObjectFileBase const&, ImporterOptions const&) /builds/gitlab/revng/orchestra/orchestra/sources/revng/lib/Model/Importer/Binary/ELFImporter.cpp:1339:1
#16 0x00007fcea78c0a22 importBinary(TupleTree<model::Binary>&, llvm::object::ObjectFile&, ImporterOptions const&) /builds/gitlab/revng/orchestra/orchestra/sources/revng/lib/Model/Importer/Binary/BinaryImporter.cpp:0:14
#17 0x00007fcea78c0b62 importBinary(TupleTree<model::Binary>&, llvm::StringRef, ImporterOptions const&) /builds/gitlab/revng/orchestra/orchestra/sources/revng/lib/Model/Importer/Binary/BinaryImporter.cpp:0:10
#18 0x00005559b1c6a158 llvm::Error::getPtr() const /builds/gitlab/revng/orchestra/orchestra/root/lib64/llvm/llvm/include/llvm/Support/Error.h:272:42
#19 0x00005559b1c6a158 llvm::Error::operator=(llvm::Error&&) /builds/gitlab/revng/orchestra/orchestra/root/lib64/llvm/llvm/include/llvm/Support/Error.h:212:18
#20 0x00005559b1c6a158 llvm::Error::Error(llvm::Error&&) /builds/gitlab/revng/orchestra/orchestra/root/lib64/llvm/llvm/include/llvm/Support/Error.h:192:11
#21 0x00005559b1c6a158 llvm::ExitOnError::operator()(llvm::Error) const /builds/gitlab/revng/orchestra/orchestra/root/lib64/llvm/llvm/include/llvm/Support/Error.h:1369:49
#22 0x00005559b1c6a158 main /builds/gitlab/revng/orchestra/orchestra/sources/revng/tools/model/import/binary/Main.cpp:48:3
#23 0x00007fcea2956083 __libc_start_main (/lib/x86_64-linux-gnu/libc.so.6+0x24083)
#24 0x00005559b1c69cd9 _start init.c:0:0
```

**Comment by Zhiwuyazhe1** [comment]
[bin.tar.gz](https://github.com/user-attachments/files/17148087/bin.tar.gz)

**Comment by aleclearmind** [comment]
Thanks for reporting this.

1. Do not use `revng model import binary`. That's deprecated.
   Use `revng analyze import-binary` if you want to squeeze out all the information from the ELF, but that's unlikely what you want to do. Using `--analyze` is usually what you're looking for.
   Is there some doc around that suggests to use it? If so, we should fix it.
2. > Unknown Encoding

   There's an unexpected input type of DWARF value in the `.eh_frame_hdr` section. It shouldn't take too much to fix it, but in the meantime you can strip away that section:

   ```
   armv7a-hardfloat-linux-uclibceabi-objcopy --remove-section .eh_frame_hdr dynamic_hello
   ```
3. As for the crash in `static_hello`, can you open a separate issue on the [revng-c](https://github.com/revng/revng-c) repo? It's entirely unrelated.
   As a quick workaround, since you seem to be only interested in the CFG, you can leave out revng-c analyses by using `--analyses-list=revng-initial-auto-analysis` instead of `--analyze`.

**Comment by Zhiwuyazhe1** [comment]
Thank you, it works!
I wii open a separate issue on the [revng-c](https://github.com/revng/revng-c) repo.

**Comment by aleclearmind** [comment]
Keeping this open until we solve the problem with `DW_EH_PE_omit`.

**Comment by Zhiwuyazhe1** [comment]
> As a quick workaround, since you seem to be only interested in the CFG, you can leave out revng-c analyses by using` --analyses-list=revng-initial-auto-analysis` instead of ` --analyze` .
If I need information about the jump table (indirect jump), do I need to drive revng-c analysis?

**Comment by aleclearmind** [comment]
> If I need information about the jump table (indirect jump), do I need to drive revng-c analysis?

No.

But please, let's keep GitHub issues for bugs. If you need general support, come visit us on [Discord](https://discord.gg/wEQtgKJxcX). Thanks.

The `DW_EH_PE_omit` is fixed in bd08864e49c0f974d2c81b3f2399a45079a3849d. Closing this.
